sql/core/src/main/scala/org/apache/spark/sql/jdbc/JDBCRDD.scala

```diff
@@ -230,13 +231,19 @@ private[sql] class JDBCRDD(
    * Turns a single Filter into a String representing a SQL expression.
    * Returns null for an unhandled filter.
    */
-  private def compileFilter(f: Filter): String = f match {
-    case EqualTo(attr, value) => s"$attr = $value"
-    case LessThan(attr, value) => s"$attr < $value"
-    case GreaterThan(attr, value) => s"$attr > $value"
-    case LessThanOrEqual(attr, value) => s"$attr <= $value"
-    case GreaterThanOrEqual(attr, value) => s"$attr >= $value"
-    case _ => null
+  private def compileFilter(f: Filter): String = {
+    def compileValue(value: Any): Any = value match {
+      case stringValue: String => s"'${escapeSql(stringValue)}'"
+      case _ => value
+    }
+    f match {
+      case EqualTo(attr, value)            => s"$attr = ${compileValue(value)}"
+      case LessThan(attr, value)           => s"$attr < ${compileValue(value)}"
+      case GreaterThan(attr, value)        => s"$attr > ${compileValue(value)}"
+      case LessThanOrEqual(attr, value)    => s"$attr <= ${compileValue(value)}"
+      case GreaterThanOrEqual(attr, value) => s"$attr >= ${compileValue(value)}"
```

Comment:
Please don't align the code vertically, this makes future changes more difficult.